pandas/_libs/tslibs/strptime.pyx

```diff
@@ -387,33 +379,30 @@ def array_strptime(
                 year = datetime_result.year
                 month = datetime_result.month
                 day = datetime_result.day
-        except ValueError:
-            if is_coerce:
-                iresult[i] = NPY_NAT
-                continue
-            raise
-        if weekday == -1:
-            weekday = date(year, month, day).weekday()
-
-        dts.year = year
-        dts.month = month
-        dts.day = day
-        dts.hour = hour
-        dts.min = minute
-        dts.sec = second
-        dts.us = us
-        dts.ps = ns * 1000
-
-        iresult[i] = npy_datetimestruct_to_datetime(NPY_FR_ns, &dts)
-        try:
+            if weekday == -1:
+                weekday = date(year, month, day).weekday()
+
+            dts.year = year
+            dts.month = month
+            dts.day = day
+            dts.hour = hour
+            dts.min = minute
+            dts.sec = second
+            dts.us = us
+            dts.ps = ns * 1000
+
+            iresult[i] = npy_datetimestruct_to_datetime(NPY_FR_ns, &dts)
             check_dts_bounds(&dts)
-        except ValueError:
+
+            result_timezone[i] = tz
+
+        except (ValueError, OutOfBoundsDatetime):
             if is_coerce:
                 iresult[i] = NPY_NAT
                 continue
-            raise
-
-        result_timezone[i] = tz
+            elif is_raise:
+                raise
+            return values, []
```

Comment:
Am I correct that interpreting that before `result` could be partially filled e.g. if the second element raised a ValueError and the first element didnt and now we would just return the input entirely?

Reply:
it would happen before as well, just one level up:

https://github.com/pandas-dev/pandas/blob/2f54a474aa754f37f82dab0184570e421207ba6d/pandas/core/tools/datetimes.py#L490-L507

The issue with doing it that way is that, for the `'coerce'` case, if there was an exception, it would set everything to `'NaT'`, rather than just the failed elements

`tslib.pyx` does the same thing:

https://github.com/pandas-dev/pandas/blob/2f54a474aa754f37f82dab0184570e421207ba6d/pandas/_libs/tslib.pyx#L638-L645

Reply:
On `main`, `1.5.2`, and on this branch we get:
```python
In [2]: to_datetime(['2020-01-01', '9999-01-01'], format='%Y-%d-%m', errors='ignore')
Out[2]: Index(['2020-01-01', '9999-01-01'], dtype='object')
```

Reply:
Okay thanks for clarifying

Reply:
happy to help, thanks for your quick review!

and please do turn down / put off requests if they're too many, I don't want to pressure you

Reply:
All good it hasn't been a problem!